Make substitution include the aarch path

maxammann opened the pull request:
###### Description of changes

On aarch64 systems the Java home is not set correctly. We are patching codeql to reference the nix JDK. Though we are creating an invalid JDK home on the aarch platform.

###### Things done

I added another sed replacement which needs to be run before the original one to take care of the `-aarch` path.

This is the original file in codeql:

```bash
arch="$(uname -m)"
if [ "$CODEQL_PLATFORM" = "osx64" ] && [ "$arch" = "arm64" ]; then
    : ${CODEQL_JAVA_HOME:=$CODEQL_DIST/tools/$CODEQL_PLATFORM/java-aarch64}
else
    : ${CODEQL_JAVA_HOME:=$CODEQL_DIST/tools/$CODEQL_PLATFORM/java}
fi

```

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

lilyinstarlight commented on pkgs/development/tools/analysis/codeql/default.nix:
```diff
@@ -34,6 +34,7 @@ stdenv.mkDerivation rec {
 
     ln -sf $out/codeql/tools/linux64/lib64trace.so $out/codeql/tools/linux64/libtrace.so
 
+    sed -i 's%\$CODEQL_DIST/tools/\$CODEQL_PLATFORM/java-aarch64%\${jdk11}%g' $out/codeql/codeql
     sed -i 's%\$CODEQL_DIST/tools/\$CODEQL_PLATFORM/java%\${jdk11}%g' $out/codeql/codeql
```
You could probably avoid a `sed` call like this, but it's a minor nit and it'll be fine either way

```suggestion
    sed -i 's%\$CODEQL_DIST/tools/\$CODEQL_PLATFORM/java\(-aarch64\)\?%\${jdk11}%g' $out/codeql/codeql
```

maxammann replied:
I think I prefer the current solution with two seds as it is clear that first the -aarch64 and then the other ocurence needs to be replaced.

lilyinstarlight reviewed (commented):
One minor nit: could you add the description to the commit message, so it's readily searchable in git log?

I'll give @jollheef a few days to provide input as maintainer, but otherwise this is a pretty simple change and looks pretty good for merge

Commit:
codeql: make substitution include the aarch path

Before this commit the path to the JDK was set wrong on aarch64. This
was because the script just appends -aarch64 on aarch64. But this is
wrong. This patch forces the script to use the nix JDK.

maxammann commented:
> One minor nit: could you add the description to the commit message, so it's readily searchable in git log?
> 

Done thanks!

maxammann commented:
Maybe @wegank or @r-ryantm can also review (guessing from the git history).

lilyinstarlight reviewed (approved):
This looks good enough to me now! (also @r-ryantm is a bot, btw)

---

Result of `nixpkgs-review pr 220337` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>codeql</li>
  </ul>
</details>
